### PR TITLE
QML test framework, add basic textedit and range editor widget tests

### PR DESCRIPTION
**Pull request by nirvn**
:drum: :drum: :drum: 

This PR introduces a new type of test covering our QML code. It's quite nice and hopefully will allow us to reduce the number of regressions introduced during development cycles.

ATM, I've only coded in two simple tests covering the textedit and range editor widgets. We can expand this at will in subsequent PRs.

Few notes:
- To be able to import QML objects outside of the /src/qml directory, I've had to add a new folder in src/qml/imports with a qmldir file to allow for QField QML objects in that src/qml directory to be imported by tests.
- I moved the invokable fieldType( QgsField ) function away from PlatformUtilities int LayerUtils, the former didn't make much sense and will make it easier to develop tests that require field details. Unfortunately it got included in the overall qml test commit, sorry :bow: 

Let's see what breaks.

### CODE REVIEW

**Comment by nirvn** [comment]
@qfield-fairy style please

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a56ef26fc0e4c4c2425d246944f9cc6c5a08ba13:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-bc23088e6d811746b13ed7131c0ea604334aa379-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-bc23088e6d811746b13ed7131c0ea604334aa379-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-bc23088e6d811746b13ed7131c0ea604334aa379-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-bc23088e6d811746b13ed7131c0ea604334aa379-x86.apk)

File: test/qml/tst_main.qml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import QtQuick 2.3
+import QtTest 1.0
+
+import QFieldControls 1.0
+
+TestCase {
```
Comment (m-kuhn) [inline test/qml/tst_main.qml:6]: Is `tst_main.qml` the correct file for widget tests?

File: test/test_qml.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/***************************************************************************
+                        test_qml_test.cpp
```
Comment (m-kuhn) [inline test/test_qml.cpp:2]:
```suggestion
                        qml_test.cpp
```

File: test/test_qml.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  public slots:
+    void qmlEngineAvailable( QQmlEngine *engine )
+    {
+      Q_UNUSED( engine )
+
+      REGISTER_SINGLETON( "org.qfield", LayerUtils, "LayerUtils" );
+
+      qRegisterMetaType<QgsField>( "QgsField" );
+      qRegisterMetaType<QVariant::Type>( "QVariant::Type" );
```
Comment (m-kuhn) [inline test/test_qml.cpp:43]: We should put that into a qml plugin which initializes things.
 -- not inside this PR

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2811fc41653199b84c6b3a8adbd0f83b5222d0e6:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1f74f236c4faaa546e976326759662dcc6cde8b8-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1f74f236c4faaa546e976326759662dcc6cde8b8-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1f74f236c4faaa546e976326759662dcc6cde8b8-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1f74f236c4faaa546e976326759662dcc6cde8b8-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ce586693782352cea0fc6302322153e2cf9b6f87:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-fb9e9c574904ad2122bad74ab49e5dc286ece1bd-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-fb9e9c574904ad2122bad74ab49e5dc286ece1bd-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-fb9e9c574904ad2122bad74ab49e5dc286ece1bd-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-fb9e9c574904ad2122bad74ab49e5dc286ece1bd-x86.apk)

File: test/test_qml_editorwidgets.cpp [hunk 4]
```diff
@@ -35,12 +33,7 @@ class Setup : public QObject
   public slots:
     void qmlEngineAvailable( QQmlEngine *engine )
     {
-      Q_UNUSED( engine )
-
-      REGISTER_SINGLETON( "org.qfield", LayerUtils, "LayerUtils" );
-
-      qRegisterMetaType<QgsField>( "QgsField" );
-      qRegisterMetaType<QVariant::Type>( "QVariant::Type" );
+      qmlInit( engine );
```
Comment (m-kuhn) [inline test/test_qml_editorwidgets.cpp:36]: Nice.
Would it make sense to share that with app?

Comment (nirvn) [reply test/test_qml_editorwidgets.cpp:36]: @m-kuhn , I was thinking about this. I was 50/50, unsure. I'm happy to implement it if you feel it's the right thing to do.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c7fb535dbbf9ac590594366319b5d18164a5be1e:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-06e45b279f1110b11bd1a64671c84665c0506312-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-06e45b279f1110b11bd1a64671c84665c0506312-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-06e45b279f1110b11bd1a64671c84665c0506312-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-06e45b279f1110b11bd1a64671c84665c0506312-x86.apk)

File: test/test_qml_editorwidgets.cpp (hunk 4 above)
Comment (m-kuhn) [reply test/test_qml_editorwidgets.cpp:36]: Let's keep it out for now, that would ideally be done as plugin init code.

File: CMakeLists.txt [hunk 5]
```diff
@@ -145,7 +145,7 @@ if (ANDROID)
 endif()
 
 set(QT_MIN_VERSION 5.14.0)
-find_package(Qt5 COMPONENTS Test Concurrent Core Qml Gui Xml Positioning Widgets Network Quick Svg OpenGL Sql Sensors WebView Bluetooth REQUIRED)
+find_package(Qt5 COMPONENTS Test Concurrent Core Qml Gui Xml Positioning Widgets Network Quick Svg OpenGL Sql Sensors WebView Bluetooth QuickTest REQUIRED)
```
Comment (m-kuhn) [inline CMakeLists.txt:148]:
```suggestion
find_package(Qt5 COMPONENTS Test Concurrent Core Qml Gui Xml Positioning Widgets Network Quick Svg OpenGL Sql Sensors WebView Bluetooth REQUIRED)

if (BUILD_TESTING)
  find_package(Qt5 COMPONENTS Test QuickTest)
endif ()
```

Should we do that?

Comment (nirvn) [reply CMakeLists.txt]: We should indeed.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for e0561aec6142b173ebc5f0ed98f1e895143b6048:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-868c25d8bc3468ea5fdd962bb6c92a3854d61a7b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-868c25d8bc3468ea5fdd962bb6c92a3854d61a7b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-868c25d8bc3468ea5fdd962bb6c92a3854d61a7b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-868c25d8bc3468ea5fdd962bb6c92a3854d61a7b-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 63255a78716dba8d4fdbc9b96e36961ba0ba7d0e:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1272695e40dd3c439d66a3616cf0bc0ff2334b09-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1272695e40dd3c439d66a3616cf0bc0ff2334b09-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1272695e40dd3c439d66a3616cf0bc0ff2334b09-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2052-1272695e40dd3c439d66a3616cf0bc0ff2334b09-x86.apk)

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**